nucliadb/nucliadb/ingest/tests/integration/orm/test_orm_resource.py

```diff
@@ -215,3 +218,140 @@ async def test_vector_duplicate_fields(
                 ), f"bad key {len(sent.vector)} {pkey1} - {pkey2} - {key}"
 
     assert count == 1
+
+
+async def test_generate_broker_message(
+    gcs_storage, maindb_driver, cache, fake_node, knowledgebox_ingest: str
+):
+    # Create a resource with all possible metadata in it
+    resource = await create_resource(gcs_storage, maindb_driver, knowledgebox_ingest)
+
+    # Now fetch it
+    async with maindb_driver.transaction() as txn:
+        kb_obj = KnowledgeBox(txn, gcs_storage, kbid=knowledgebox_ingest)
+        r = await kb_obj.get(resource.uuid)
+        assert r is not None
+
+    async with maindb_driver.transaction() as txn:
+        r.txn = txn
+        bm = await r.generate_broker_message()
+
+    # Check generated broker message has the same metadata as the created resource
+
+    # 1. ROOT ELEMENTS
+    # 1.1 BASIC
+    basic = bm.basic
+    assert basic.HasField("created")
```

Comment:
that's a huge list of asserts. I wonder if you could create a `basic` object and assert it just once?

Reply:
I know... the problem is that the basic has some datetime fields that will make the assertion false :(